projects/kalidea/kaligraphi/src/lib/02-form/kal-select-virtual-scroll/kal-select-virtual-scroll.component.ts

```diff
@@ -0,0 +1,297 @@
+      return;
+    }
+
+    this.focus();
+
+    if (!this.overlayRef) {
+      this.createOverlay();
+    }
+    this.overlayRef.attach(this.optionsPortal);
+    this.isPanelOpen = true;
+  }
+
+  close(): void {
+    if (this.overlayRef){
+      this.overlayRef.detach();
+    }
+    this.isPanelOpen = false;
+  }
+
+  /**
+   * create overlayRef
+   */
+  private createOverlay() {
+    const positionStrategy = this.overlay
+      .position()
+      .flexibleConnectedTo(this.elementRef)
+      .withPositions([
+        {originX: 'start', originY: 'bottom', overlayX: 'start', overlayY: 'top'}
+      ]);
+
+    this.overlayRef = this.overlay.create({
+      positionStrategy,
+      hasBackdrop: true,
+      backdropClass: 'kal-overlay__transparent',
+      width: this.elementRef.nativeElement.getBoundingClientRect().width,
+      scrollStrategy: this.overlay.scrollStrategies.reposition()
+    });
+
+    this.overlayRef.backdropClick().subscribe(() => {
+      this.isFocused = false;
+      this.close();
+    });
+
+    this.overlayRef.keydownEvents()
+      .pipe(filter(event => event.keyCode === ESCAPE))
+      .subscribe(() => this.close());
+  }
+
+  isActive(option) {
+    if (!this.selected) {
+      return false;
+    }
+    return option[this.idKey] === this.selected[this.idKey];
+  }
+
+  get label() {
+    return this.selected ? this.selected[this.labelKey] : 'Select...';
+  }
+
+  private setOptions( dataSource$: Observable<T[] | ReadonlyArray<T>>) {
+    this.dataSourceSubscription = combineLatest( dataSource$, this.searchControl.valueChanges
+      ).pipe(
+        map( ([items, term]: [T[], string]) => {
+          console.log(`filtering options ${items} with ${term}`);
+          return items.filter( item => item.label.includes(term));
+        })
+      ).subscribe(
+      (items: T[] ) => {
+        this.options = items !== undefined ? items : [];
+        this.cdr.markForCheck();
+        console.log(this.options);
+      }
+    );
+  }
+
+  private observeDataSource() {
+    if ((this.dataSource as DataSource<T>).connect instanceof Function) {
+      this.setOptions( (this.dataSource as DataSource<T>).connect(this));
+    } else if ( this.dataSource instanceof Observable) {
+      this.setOptions((this.dataSource as Observable<T[]>));
+    } else if ( Array.isArray(this.dataSource)) {
+      this.setOptions(of(this.dataSource as T[]));
+    }
+  }
+
+  private destroyDataSourceSubscription() {
+    this.selected = null;
+    this.dataSourceSubscription.unsubscribe();
+
+    if (this.dataSource && (this.dataSource as DataSource<T>).connect instanceof Function){
+      (this.dataSource as DataSource<T>).disconnect(this);
+    }
+  }
+
+  ngOnInit() {
+    // this.originalOptions = [...this.options];
+    if (this.selected !== undefined) {
+      this.selected = this.options.find(
+        currentOption => currentOption[this.idKey] === this.selected
+      );
+    }
+
+    this.searchControl.setValue('');
```

Comment:
You can remove the `setValue` and set a default value when declaring the control : `searchControl = new FormControl('');`